client/alloc_runner.go

```diff
@@ -344,7 +343,6 @@ OUTER:
 		case update := <-r.updateCh:
 			// Check if we're in a terminal status
 			if update.TerminalStatus() {
-				r.setAlloc(update)
```

Comment:
Removed the method as well. I removed it because there was a race between this setting the status and the `dirtySyncState` when the user stops the job. It would trigger an update after the destroy so the incorrect status would be set.